app/debrief_gui.py

```diff
@@ -90,105 +108,94 @@ async def download_json(self, request):
-            if sections['facts-table']:
-                story_obj.append_text('FACTS FOUND IN OPERATION <font name=Courier-Bold size=17>%s</font>' % o.name.upper(),
-                                      styles['Heading2'], 0)
-                story_obj.append_text(story_obj.get_description('op facts'), styles['Normal'], 12)
-                story_obj.append(story_obj.generate_facts_found(o))
-                story_obj.page_break()
-
-        # pdf teardown
-        doc.build(story_obj.story_arr,
-                  onFirstPage=story_obj.header_footer_first,
-                  onLaterPages=story_obj.header_footer_rest)
+
+        # content generation
+        try:
+            for section in sections:
+                section_module = self.report_section_modules.get(section, None)
+                if section_module:
+                    flowables = section_module.generate_section_elements(
+                        styles,
+                        operations=operations,
+                        agents=agents,
+                        graph_files=graph_files,
+                    )
+                    for flowable in flowables:
+                        story_obj.append(flowable)
+                else:
+                    self.log.error("Requested debrief section module %s not found." % section)
+
+            # pdf teardown
+            doc.build(story_obj.story_arr,
+                      onFirstPage=story_obj.header_footer_first,
+                      onLaterPages=story_obj.header_footer_rest)
+        except Exception as e:
+            self.logger.error(e)
         pdf_value = pdf_buffer.getvalue()
         pdf_buffer.close()
         return pdf_value.decode('utf-8', errors='ignore')
 
     @staticmethod
-    def _generate_ttps(operations):
-        ttps = dict()
-        for op in operations:
-            for link in op.chain:
-                if not link.cleanup:
-                    tactic_name = link.ability.tactic
-                    if tactic_name not in ttps.keys():
-                        tactic = dict(name=tactic_name,
-                                      techniques={link.ability.technique_name: link.ability.technique_id},
-                                      steps={op.name: [link.ability.name]})
-                        ttps[tactic_name] = tactic
-                    else:
-                        if link.ability.technique_name not in ttps[tactic_name]['techniques'].keys():
-                            ttps[tactic_name]['techniques'][link.ability.technique_name] = link.ability.technique_id
-                        if op.name not in ttps[tactic_name]['steps'].keys():
-                            ttps[tactic_name]['steps'][op.name] = [link.ability.name]
-                        elif link.ability.name not in ttps[tactic_name]['steps'][op.name]:
-                            ttps[tactic_name]['steps'][op.name].append(link.ability.name)
-        return dict(sorted(ttps.items()))
+    def _sanitize_filename(filename):
```

Comment:
let's move to a utility function within base caldera (this pr or later)

Reply:
Maybe a later PR for now, in case we want to agree on a standardized file sanitization method for base Caldera